challenge_10/java/zmiller91/ValidClosers.java

```diff
@@ -0,0 +1,68 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Stack;
+
+/**
+ * @author zmiller
+ */
+public class ValidClosers extends YOP {
+    
+    private final String FALSE = "false";
+    private final String TRUE = "true";
+    private final List<Character> openers = Arrays.asList('{', '[', '(');
+    private final List<Character> closers = Arrays.asList('}', ']', ')');
+    
+    /**
+     * Validates that the given input has a correct sequence of openers and 
+     * closers
+     * 
+     * @param input 
+     */
+    protected void run(String input) {
+        
+        Stack<Character> stack = new Stack<>();
+        for(char c : input.toCharArray()) {
+            
+            // No rules for openers, add it to the stack
+            if (openers.contains(c)) {
+                stack.push(c);
+            }
+            
+            // Openers have a few rules
+            if (closers.contains(c)) {
+                
+                // Stack cannot be empty
+                if(stack.empty()) {
+                    System.out.println(FALSE);
```

Comment:
I put it as a variable for a reason -- it allows me to change the string to "Sorry, please try again" with ease. It was an intentional design decision.